cowprotocol/services
# Support JIT orders in the trade verifier

# Description
Closes task n2 from https://github.com/cowprotocol/services/issues/3082 by implementing support of the quotes with JIT orders in the trade verifier. For a gradual migration, it has to support both quote versions.

# Changes

- [ ] Added a new version of the trade with JIT orders.
- [ ] Utilized the clearing prices to calculate the out amount.
- [ ] Altered the `Solver.sol` helper contract to fetch all token balances as was proposed in [one of the comments](https://github.com/cowprotocol/services/pull/3085#discussion_r1830692576), which reduces the overall code complexity.
- [ ] Bumped into an issue while converting floats into `BigRational`. ~~Implemented a workaround with converting float's string representation into `BigRational`.~~ Used `BigDecimal` in the config instead.

## How to test
Unit tests. e2e would be possible only once the driver support is implemented(see https://github.com/cowprotocol/services/pull/3103).
